plasmapy/diagnostics/charged_particle_radiography/synthetic_radiography.py

```diff
@@ -1047,137 +843,65 @@ def run(
         have left the grid, they are advanced to the
         detector plane where they can be used to construct a synthetic
         diagnostic image.
-
-        Parameters
-        ----------
-        dt : `~astropy.units.Quantity`, optional
-            An explicitly set timestep in units convertible to seconds.
-            Setting this optional keyword overrules the adaptive time step
-            capability and forces the use of this timestep throughout. If a tuple
-            of timesteps is provided, the adaptive timestep will be clamped
-            between the first and second values.
-
-        field_weighting : str
-            String that selects the field weighting algorithm used to determine
-            what fields are felt by the particles. Options are:
-
-            * 'nearest neighbor': Particles are assigned the fields on
-                the grid vertex closest to them.
-            * 'volume averaged' : The fields experienced by a particle are a
-                volume-average of the eight grid points surrounding them.
-
-            The default is 'volume averaged'.
-
-        Returns
-        -------
-        None
-
         """
 
-        # Load and validate inputs
-        field_weightings = ["volume averaged", "nearest neighbor"]
-        if field_weighting in field_weightings:
-            self.field_weighting = field_weighting
-        else:
-            raise ValueError(
-                f"{field_weighting} is not a valid option for ",
-                "field_weighting. Valid choices are",
-                f"{field_weightings}",
-            )
-
-        # By default, set dt as an infinite range (auto dt with no restrictions)
-        self.dt = np.array([0.0, np.inf]) * u.s if dt is None else dt
-        self.dt = (self.dt).to(u.s).value
-
-        # Check to make sure particles have already been generated
-        if not hasattr(self, "x"):
-            raise ValueError(
-                "Either the create_particles or load_particles method must be "
-                "called before running the particle tracing algorithm."
-            )
+        self._validate_inputs(field_weighting)
 
         # If meshes have been added, apply them now
         for mesh in self.mesh_list:
             self._apply_wire_mesh(**mesh)
 
-        # Store a copy of the initial velocity distribution in memory
-        # This will be used later to calculate the maximum deflection
-        self.v_init = np.copy(self.v)
```

Comment:
This line should stay here (only store the init arrays for the charged particle radiography case)